packages/realm/src/tests/schema-normalization.test.ts

```diff
@@ -17,59 +17,846 @@
+    // User-defined type: Person
+    // -------------------------
+
+    itNormalizes(
+      {
+        type: "object",
+        objectType: "Person",
+      },
+      {
+        type: "object",
+        objectType: "Person",
+        optional: true,
+      },
+    );
+
+    itNormalizes(
+      {
+        type: "object",
+        objectType: "Person",
+        optional: true,
+      },
+      {
+        type: "object",
+        objectType: "Person",
+        optional: true,
+      },
+    );
+
+    itNormalizes(
+      {
+        type: "list",
+        objectType: "Person",
+      },
+      {
+        type: "list",
+        objectType: "Person",
+        optional: false,
+      },
+    );
+
+    itNormalizes(
+      {
+        type: "list",
+        objectType: "Person",
+        optional: false,
+      },
+      {
+        type: "list",
+        objectType: "Person",
+        optional: false,
+      },
+    );
+
+    itNormalizes(
+      {
+        type: "set",
+        objectType: "Person",
+      },
+      {
+        type: "set",
+        objectType: "Person",
+        optional: false,
+      },
+    );
+
+    itNormalizes(
+      {
+        type: "set",
+        objectType: "Person",
+        optional: false,
+      },
+      {
+        type: "set",
+        objectType: "Person",
+        optional: false,
+      },
+    );
+
+    itNormalizes(
+      {
+        type: "dictionary",
+        objectType: "Person",
+      },
+      {
+        type: "dictionary",
+        objectType: "Person",
+        optional: true,
+      },
+    );
+
+    itNormalizes(
+      {
+        type: "dictionary",
+        objectType: "Person",
+        optional: true,
+      },
+      {
+        type: "dictionary",
+        objectType: "Person",
+        optional: true,
+      },
+    );
+
+    itNormalizes(
+      {
+        type: "linkingObjects",
+        objectType: "Person",
+        property: "tasks",
+      },
+      {
+        type: "linkingObjects",
+        objectType: "Person",
+        property: "tasks",
+        optional: false,
+      },
+    );
+
+    itNormalizes(
+      {
+        type: "linkingObjects",
+        objectType: "Person",
+        property: "tasks",
+        optional: false,
+      },
+      {
+        type: "linkingObjects",
+        objectType: "Person",
+        property: "tasks",
+        optional: false,
+      },
+    );
   });
 
-  it("transforms a string declaring a list of class name", () => {
-    const result = normalizePropertySchema("prop", "Person[]");
-    expect(result.name).equals("prop");
-    expect(result.type).equals("list");
-    expect(result.objectType).equals("Person");
-    expect(result.optional).equals(false); // Lists of objects may never be optional
+  // ------------------------------------------------------------------------
+  // Invalid object notation
+  // ------------------------------------------------------------------------
+
+  describe("using invalid object notation", () => {
```

Comment:
Add a test for `itThrowsWhenNormalizing({}, ...);`

Reply:
Thanks 👍

Reply:
Update: Will be added to the validation tests.